Add Order Amend section to the guide

james-harper opened the pull request:
### Added
- Added an order amend section to the guide

[Delivers [#141296853](https://www.pivotaltracker.com/n/projects/1404974/stories/141296853)]

wnowicki commented on _includes/v4/applications.md:
```diff
@@ -175,3 +175,20 @@ Name | Required | Type | Description
+Name | Type | Description
+--- | --- | ---
+`amount`| int | Payment amount in pence (must be `>0`)
+`description` | string | Short description of the [`order`](#order) being amended
```
- what will happen with this description? is it a new order description to replace previous one?

james-harper commented:
@wnowicki I've added a section for features on the main retailer integration page (just contains order amend for now) that describes what the feature does and highlights how action is required to enable it.

Also added more information to the order amend section on the api page

wnowicki commented:
Closing as **Order Amend** approach has changed
